R/plot-construction.R

```diff
@@ -87,49 +87,80 @@ add_ggplot <- function(p, object, objectname) {
 #'
 #' @keywords internal
 #' @export
-ggplot_add <- function(object, plot, object_name) {
-  UseMethod("ggplot_add")
-}
-#' @export
-ggplot_add.default <- function(object, plot, object_name) {
+ggplot_add <- new_generic(
+  "ggplot_add",
+  dispatch_args = c("object", "plot"),
+  fun = function(object, plot, object_name) S7_dispatch()
```

Comment:
I'd be tempted to leave `object_name` as an optional argument for the methods (i.e. just rely on the default constructor which will also include `...`, which I'd now consider to be best practice in a generic. OTOH it might be better to make that change in a separate PR, as I'd probably now call this `error_arg`, and I'd expect a matching `error_call` argument in order to apply our current error best practices.

Reply:
Fair point, I'll leave this comment unresolved to not forget about this for now.